src/vs/base/common/comparers.ts

```diff
@@ -6,30 +6,168 @@
 import { sep } from 'vs/base/common/path';
 import { IdleValue } from 'vs/base/common/async';
 
-const intlFileNameCollator: IdleValue<{ collator: Intl.Collator, collatorIsNumeric: boolean }> = new IdleValue(() => {
-	const collator = new Intl.Collator(undefined, { numeric: true, sensitivity: 'base' });
+// When comparing large numbers of strings, such as in sorting large arrays, is better for
+// performance to create an Intl.Collator object and use the function provided by its compare
+// property than it is to use String.prototype.localeCompare()
+
+// A collator with numeric sorting enabled.
+const intlFileNameCollatorNumeric: IdleValue<{ collator: Intl.Collator }> = new IdleValue(() => {
+	const collator = new Intl.Collator(undefined, { numeric: true });
+	return {
+		collator: collator
+	};
+});
+
+// A collator with numeric sorting enabled, and sensitivity to accents but not case.
+const intlFileNameCollatorNumericCaseInsenstive: IdleValue<{ collator: Intl.Collator }> = new IdleValue(() => {
+	const collator = new Intl.Collator(undefined, { numeric: true, sensitivity: 'accent' });
 	return {
-		collator: collator,
-		collatorIsNumeric: collator.resolvedOptions().numeric
+		collator: collator
 	};
 });
 
-export function compareFileNames(one: string | null, other: string | null, caseSensitive = false): number {
-	const a = one || '';
-	const b = other || '';
-	const result = intlFileNameCollator.getValue().collator.compare(a, b);
```

Comment:
Note that unlike the other comparisons, this one wasn't separately comparing names and extensions - which cause problems like the `aggregate_repo.go` vs `aggregate.go` issue.